fiftyone/core/labels.py

```diff
@@ -1278,6 +1278,62 @@ def to_image_labels(self, name=None):
         """
         return etai.ImageLabels(mask=self.mask, tags=self.tags)
 
+    def to_detections(self, mask_targets=None):
```

Comment:
Is there a use case for this?

If I wanted to do a transformation like this, I would likely want the default (or at least an option) to create one detection per connected region rather than always one per class. In fact, I could see the user wanting to be able to specify this on a per-class basis ("stuff-like" vs "thing-like" distinction used in some papers).

I might also consider the name `to_instances()` for this method, since, although  the data type returned is `Detections`, as it is a bit more specific.

But, in any case, first question: how this is used right now?

Reply:
**Current use case**: CVAT processes all segmentations as polygons so I needed a way to go from `Segmentation` to `Polyline`. Since detections already had a `to_polyline()` method, I just converted from `Segmentation` to `Detections` and then go to polylines from there.

The fact that it creates one instance per class is fine since that is how it is annotated in CVAT, so splitting it into instances would just be unnecessary computation for this use case.

I agree that this isn't the most useful method otherwise. Also `Segmentations.to_polylines()` may be preferable, but looking through the code it didn't seem like it would be too much more efficient than `Segmentations` -> `Detections` -> `Polylines` so I just left it as is.

Reply:
I see. Just because I like useful utilities, I think I'll take a whack at upgrading this method to allow for customizing stuff vs thing-like behavior, and the CVAT logic can use stuff-only.

I'll probably throw in `to_polylines()` as well, for completeness.